actors/builtin/reward/reward_actor.go

```diff
@@ -93,12 +93,15 @@ func (a Actor) AwardBlockReward(rt vmr.Runtime, params *AwardBlockRewardParams)
 	AssertMsg(big.Add(rewardPayable, penalty).LessThanEqual(priorBalance),
 		"reward payable %v + penalty %v exceeds balance %v", rewardPayable, penalty, priorBalance)
 
+	// if this fails, we can assume the miner is responsible and do not to fail here
 	_, code := rt.Send(minerAddr, builtin.MethodsMiner.AddLockedFund, &rewardPayable, rewardPayable)
-	builtin.RequireSuccess(rt, code, "failed to send reward to miner: %s", minerAddr)
+	if !code.IsSuccess() {
+		rt.Log(vmr.ERROR, "failed to send AddLockedFund call to the miner actor with funds: %v, code: %v", rewardPayable, code)
```

Comment:
@anorth should we turn move the reward to the penalty if paying the miner fails?

Reply:
Sorry, I can't parse your question.

Reply:
(proofreading...)

If sending funds to a miner fails, what should we do with them? At the moment, it looks like the reward actor will just accrue funds. Given the fact that we're intentionally burning penalty below, I assume that any funds _not_ sent to the miner should also be burnt.

Reply:
Yes, ok let's do that.

The reward actor accrues unpaid block rewards when the election winner just fails to get a block into the chain at all. But that maintains a property of the reward actor's balance decreasing exactly as expected given the blocks actually in the chain. We can retain that here by burning the reward (and also the gas reward that was sent here).